app/models/component.rb

```diff
@@ -71,9 +71,16 @@ def add_intermediate_file(structure, filegrp, div)
     structure.add_fptr(parent: div, fileid: file['ID'])
   end
 
-  def add_service_file(structure, filegrp, div)
```

Comment:
I feel as though these `add_*_file` private methods should be public methods of `Ddr::Models::Structure`.